app/controllers/users_controller.rb

```diff
@@ -1,5 +1,7 @@
 class UsersController < ApplicationController
   before_action :find_user, except: %i(create index new)
+  before_action :require_admin_or_self, only: %i(edit update)
+  before_action :require_admin_in_roster, except: %i(edit update)
```

Comment:
Here too, albeit not quite as clearly, this could be:

```ruby
before_action :require_admin_in_roster_or_self
before_action :require_admin_in_roster, except: %i(edit update)
```